crates/interpreter/tests/spec.rs

```diff
@@ -24,44 +28,79 @@ use wast::lexer::Lexer;
 use wast::token::Id;
 use wast::{parser, QuoteWat, Wast, WastArg, WastDirective, WastExecute, WastInvoke, WastRet, Wat};
 
-fn test(repo: &str, name: &str) {
-    let path = format!("../../third_party/WebAssembly/{repo}/test/core/{name}.wast");
-    let content = std::fs::read_to_string(path).unwrap();
-    let mut lexer = Lexer::new(&content);
-    lexer.allow_confusing_unicode(true);
-    let buffer = parser::ParseBuffer::new_with_lexer(lexer).unwrap();
-    let wast: Wast = parser::parse(&buffer).unwrap();
+fn parse_directives(name: &str, directives: Vec<WastDirective>, root_env: Option<Arc<Env>>) {
     let layout = std::alloc::Layout::from_size_align(pool_size(name), MEMORY_ALIGN).unwrap();
-    let pool = unsafe { std::slice::from_raw_parts_mut(std::alloc::alloc(layout), layout.size()) };
-    let mut env = Env::new(pool);
-    env.instantiate("spectest", &SPECTEST);
-    env.register_name("spectest", None);
-    assert!(env.inst.is_ok());
-    for directive in wast.directives {
+    let pool =
+        unsafe { std::slice::from_raw_parts_mut(std::alloc::alloc_zeroed(layout), layout.size()) };
+    let mut env_arc = Arc::new(Env::new(pool));
+    let first_call = root_env.is_some();
+    if root_env.is_some() {
+        env_arc = root_env.unwrap();
+    }
+    let env_arc_cpy = env_arc.clone();
+    let env = unsafe { Arc::<Env<'_>>::get_mut_unchecked(env_arc.borrow_mut()) };
```

Comment:
This is actually unsound. When we spawn multiple threads, they will each get a mutable reference to the environment.